placement/algo/mirrored.go

```diff
@@ -285,6 +267,83 @@ func allInstancesInState(
 
 	return len(instanceIDs) == 0
 }
+
+func (a mirroredAlgorithm) returnInitializingShards(
+	p placement.Placement,
+	instanceIDs []string,
+) (placement.Placement, error) {
+	shouldContinue := true
```

Comment:
nit: take it or leave it, but `progressed` or `madeProgress` might be easier to understand and more specific than `shouldContinue`.